### DISCUSSION TITLE
RustDesk iOS - Allow to use native keyboard when connecting to mobile device

### DISCUSSION THREAD

**Discussion by plia7** [rustdesk/rustdesk#7183] [Feature Request]
Hello,

When I connect to an Android target using RustDesk iOS app, the keyboard icon appears to be missing from the menu. It does show if I connect to a windows based target.

No keyboard icon when connecting to mobile device:

![BBB25C38-A029-46B8-8A08-1ED789A1D4C0](https://github.com/rustdesk/rustdesk/assets/134864072/680cd046-0165-40a3-b475-340072fade9b)

For reference, here is the keyboard icon I’m talking about that shows if I connect to a Windows based target:

![6809AB38-0297-459A-8D4E-55F45DFC1A22](https://github.com/rustdesk/rustdesk/assets/134864072/230b557f-7037-4e0d-912d-6c640551f0aa)

Could you please allow to use native keyboard when connecting to mobile device?

I tried to use the remote soft keyboard of Android, but It's too slow and prone to errors to try to type on the target Android soft keyboard that’s on the target screen rather than sending the keystrokes from the client connecting iOS side native soft keyboard.

Other remote desktop apps including TeamViewer allow this without any problem. Here is an example for another remote desktop app that gives you the option to either use the target android soft keyboard on screen or the source ios soft keyboard:

https://imgur.com/a/m5OirEK

I believe that’s how it should be in RustDesk too. The way it is right now basically makes the keyboard input very difficult and extremely unreliable especially if facing with a less stable connection.

I hope the development team will consider my request. I think it’s a real and serious issue that needs to be addressed to improve the software.

Thank you.

Add the keyboard icon to show when connecting to android and then open the native iOS on screen keyboard to show so you don't need to rely on a slow/non-reliable android on screen keyboard which is also blocked by some RustDesk widgets such as the blue up arrow and RustDesk toolbar? If I want to use the keyboard, I should not have RustDesk widgets blocking the view of the on screen keyboard.

![image](https://github.com/rustdesk/rustdesk/assets/134864072/91a63abf-ba76-42fd-aebd-266841cf9e76)

![image](https://github.com/rustdesk/rustdesk/assets/134864072/53fcb0e7-8d05-4721-83fa-3aed6b18fffa)

**Comment by plia7** [comment]
up

**Comment by plia7** [comment]
@fufesou Could you please implement the functionality? Add the keyboard icon to show when connecting to android and then open the native iOS on screen keyboard to show so you don't need to rely on a slow/non-reliable android on screen keyboard which is also blocked by some RustDesk widgets such as the blue up arrow and RustDesk toolbar? If I want to use the keyboard, I should not have RustDesk widgets blocking the view of the on screen keyboard.

![image](https://github.com/rustdesk/rustdesk/assets/134864072/91a63abf-ba76-42fd-aebd-266841cf9e76)

![image](https://github.com/rustdesk/rustdesk/assets/134864072/53fcb0e7-8d05-4721-83fa-3aed6b18fffa)

**Reply by fufesou** [reply]
Good suggestion!

But we have some other functions to do. It depends on @rustdesk 's plan.

**Comment by rustdesk** [comment]
Lock it. Will unlock once we find time to handle this.

We have our plan, stop pushing.
